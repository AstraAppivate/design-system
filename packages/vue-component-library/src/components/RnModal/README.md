---
title: Modal
description: Modals are empty components that cover the user's screen.
---

# Basic Example

```html
  <rn-modal @click="close" title="Example Title"></rn-modal>
```

#### Props

Prop Name     | Value      | Required
------------- | ---------- | --------
**title**     | `{String}` | false
