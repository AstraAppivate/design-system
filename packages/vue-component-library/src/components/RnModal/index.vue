<template lang="html">
  <div class="rn-modal" :class="{ error: error }">
    <div class="window">
      <div class="header" v-if="title">
        <h4 class="title">{{ title }}</h4>
      </div>
      <div class="content">
        <slot></slot>
      </div>
      <div class="footer">
        <template v-if="!hideCancel">
          <rn-button class="secondary h_mr-2" type="link" @click="$emit('close')">Cancel</rn-button>
        </template>
        <template v-if="!hideAction">
          <rn-button v-if="error" type="error" @click="$emit('clickAction')">{{ actionButtonText }}</rn-button>
          <rn-button state="neutral regular" v-else @click="$emit('clickAction')">{{ actionButtonText }}</rn-button>
        </template>
      </div>
    </div>
  </div>
</template>

<script src="./scripts.js"></script>
