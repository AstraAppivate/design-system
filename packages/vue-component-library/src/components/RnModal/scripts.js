import RnButton from '../RnButton/index.vue'
import RnCard from '../RnCard/index.vue'

import '@royalnavy/css-framework/src/components/_modal.scss'

export default {
  name: 'RnModal',
  components: {
    RnCard,
    RnButton,
  },
  data: () => ({
    isOpen: false,
  }),
  props: {
    title: String,
    hideCancel: {
      type: Boolean,
      default: false,
    },
    hideAction: {
      type: Boolean,
      default: false,
    },
    actionButtonText: {
      type: String,
      default: 'OK',
    },
    error: Boolean,
  },
}
