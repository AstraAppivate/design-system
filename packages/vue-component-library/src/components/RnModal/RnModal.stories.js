import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import RnModal from './index.vue'
import RnButton from '../RnButton/index.vue'

storiesOf('Modals', module)
  .add('Basic', () => ({
    components: { RnModal, RnButton },
    data() {
      return {
        isOpen: false
      }
    },
    template: `
    <div>
      <rn-button style="position:relative;z-index:9999" id="show-modal" state="neutral regular" @click="isOpen = true">Activate Modal</rn-button>
      <rn-modal
        title="Generic Card Content with a long title that wraps multiple rows"
        actionButtonText="I Understand"
        :class="{ open: isOpen }"
        @close="isOpen = false"
        @clickAction="isOpen = false"
      >
        <p>Sed posuere consectetur est at lobortis. Curabitur blandit tempus porttitor. Curabitur blandit tempus porttitor. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Donec sed odio dui.</p>
      </rn-modal>
    </div>
    `,
  }))
  .add('Hide action', () => ({
    components: { RnModal, RnButton },
    data() {
      return {
        isOpen: false
      }
    },
    template: `
    <div>
      <rn-button style="position:relative;z-index:9999" id="show-modal" state="neutral regular" @click="isOpen = true">Activate Modal</rn-button>
      <rn-modal
        error
        hideAction
        title="Generic Card Content with a long title that wraps multiple rows"
        :class="{ open: isOpen }"
        @close="isOpen = false"
      >
        <p>Sed posuere consectetur est at lobortis. Curabitur blandit tempus porttitor. Curabitur blandit tempus porttitor. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Donec sed odio dui.</p>
      </rn-modal>
    </div>
      `,
  }))
  .add('Hide cancel', () => ({
    components: { RnModal, RnButton },
    data() {
      return {
        isOpen: false
      }
    },
    template: `
    <div>
      <rn-button style="position:relative;z-index:9999" id="show-modal" state="neutral regular" @click="isOpen = true">Activate Modal</rn-button>
      <rn-modal
        error
        hideCancel
        title="Generic Card Content with a long title that wraps multiple rows"
        :class="{ open: isOpen }"
        @clickAction="isOpen = false"
      >
        <p>Sed posuere consectetur est at lobortis. Curabitur blandit tempus porttitor. Curabitur blandit tempus porttitor. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Donec sed odio dui.</p>
      </rn-modal>
    </div>
      `,
  }))
  .add('No Title', () => ({
    components: { RnModal, RnButton },
    data() {
      return {
        isOpen: false
      }
    },
    template: `
    <div>
      <rn-button style="position:relative;z-index:9999" id="show-modal" state="neutral regular" @click="isOpen = true">Activate Modal</rn-button>
      <rn-modal
        actionButtonText="OK"
        :class="{ open: isOpen }"
        @close="isOpen = false"
        @clickAction="isOpen = false"
      >
        <p>Sed posuere consectetur est at lobortis. Curabitur blandit tempus porttitor. Curabitur blandit tempus porttitor. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Donec sed odio dui.</p>
      </rn-modal>
    </div>
    `,
  }))
