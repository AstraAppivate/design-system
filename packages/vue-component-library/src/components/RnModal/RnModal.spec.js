import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import RnModal from '@/src/components/RnModal/index.vue'

describe('RnModal.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(RnModal, {
      propsData: {
        //
      },
    })
  })

  it('snapshot: has same HTML structure', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err)
      }

      expect(str).toMatchSnapshot()
    })
  })
})
