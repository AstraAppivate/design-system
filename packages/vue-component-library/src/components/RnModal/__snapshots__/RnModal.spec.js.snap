// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnModal.vue snapshot: has same HTML structure 1`] = `
<div class="rn-modal">
  <div class="window">
    <!---->
    <div class="content"></div>
    <div class="footer">
      <rn-button-stub type="link" class="secondary h_mr-2">Cancel</rn-button-stub>
      <rn-button-stub state="neutral regular">OK</rn-button-stub>
    </div>
  </div>
</div>
`;
